articles/ros2_dds_security.md

```diff
@@ -125,62 +125,63 @@ Let's delve further into these.
 
 #### Directory tree of all security files
 
-RCL supports finding security files in one directory that is the root of a directory structure corresponding to the fully-qualified names of every node instance (i.e. namespace + node name).
-For example, for the `/front/camera` node, the directory structure would look like:
+RCL supports finding security files in one directory that is the root of a directory structure corresponding to the fully-qualified names of every context (e.g. namespace + context name).
+For example, for the `/front/camera` context, the directory structure would look like:
 
     <root>
-    └── front
-        └── camera
-            ├── cert.pem
-            ├── key.pem
-            ├── ...
+    └── contexts
```

Comment:
This structure isn't quite what I see reflected in [the RCL PR](https://github.com/ros2/rcl/pull/515/files#diff-66385e777ba2ba5cee4777a905befaf0R40). What purpose does the `contexts` directory serve?

Reply:
So that we don't just blast all the context paths in the root of the keystore. For the same reason we using plural element names like `contexts` or `profiles` in the XML schema, so that we can easily extend features in an orderly manner.

Another reason, using apparmor, how would one allow read access to only intended public certs in the root of the keystore, but not the private certs. Splitting out the directories makes it simpler to write access rules, like for Linux Security Modules.

Reply:
Yeah we'll need to update the RCL PR to match the behavior described in this doc (also applies to behavior for not updated rmws etc).

The question is valid though, an extention of it is:
 is it better to ask users to point "ROS_SECURE_ROOT_DIRECTORY" to `/path/to/keystore` and add the `contexts` artificially? or just ask users to set  "ROS_SECURE_ROOT_DIRECTORY" to `/path/to/keystore/contexts` directly ?

I'd prefer the second one but I guess the variable name would not really be appropriate anymore

Reply:
>I'd prefer the second one but I guess the variable name would not really be appropriate anymore

I think we should ask ourselves if we foresee anything other than the contexts as needed to be co-located in the keystore root. The public folder is one example, but there could be more in the future, such as meta files for formal verification like caching proof results, or additional keystore layouts for working with TPM devices, or auto generated policy files from introspective tooling.

This was why with Keymint, I went with a workspace like layout with `public`, `private`, `src`, `build` and `install`, directing the user to use the `install` folder for looking up artifacts: https://github.com/keymint/keymint_tools#example

Reply:
> The question is valid though, an extention of it is:
is it better to ask users to point "ROS_SECURE_ROOT_DIRECTORY" to /path/to/keystore and add the contexts artificially? or just ask users to set "ROS_SECURE_ROOT_DIRECTORY" to /path/to/keystore/contexts directly ?

In the second case, I think we won't need to update the rcl PR (except changing the variable name).

Reply:
> just ask users to set "ROS_SECURE_ROOT_DIRECTORY" to /path/to/keystore/contexts directly 

The design doc has been updated to just use this convention, see https://github.com/ros2/design/pull/274/commits/fc40702c6f704e54deae4919937a5312f3ccea26#diff-0c022ad3d42831e4e8cf2336da70d354R99